laws/src/main/scala/cats/laws/FunctorLaws.scala

```diff
@@ -33,6 +33,15 @@ trait FunctorLaws[F[_]] extends InvariantLaws[F] {
   def covariantIdentity[A](fa: F[A]): IsEq[F[A]] =
     fa.map(identity) <-> fa
 
+  def mapOrKeepEmpty[A](fa: F[A]): IsEq[F[A]] =
+    fa.mapOrKeep(PartialFunction.empty) <-> fa
+
+  def mapOrKeepIdentity[A](fa: F[A]): IsEq[F[A]] =
+    fa.mapOrKeep { case a => a } <-> fa
```

Comment:
Maybe you want to take a function from A to A here and make it the same as map?

Reply:
updated